packages/react-components/react-virtualizer/src/hooks/useVirtualizerMeasure.ts

```diff
@@ -97,3 +97,140 @@ export const useStaticVirtualizerMeasure = (
+    }
+  }
+
+  const [state, setState] = React.useState({
+    virtualizerLength: 0,
+    virtualizerBufferItems: 0,
+    virtualizerBufferSize: 0,
+  });
+
+  const { virtualizerLength, virtualizerBufferItems, virtualizerBufferSize } = state;
+
+  const container = React.useRef<HTMLElement | null>(null);
+
+  const resizeCallback = () => {
+    if (!container.current) {
+      // Error? ignore?
+      return;
+    }
+
+    const containerSize =
+      direction === 'vertical'
+        ? container.current?.getBoundingClientRect().height
+        : container.current?.getBoundingClientRect().width;
+
+    let indexSizer = 0;
+    let length = 0;
+
+    while (indexSizer <= containerSize) {
+      const iItemSize = getItemSize(currentIndex + length);
+      sizeTracker.current[currentIndex + length] = iItemSize;
+
+      // Increment
+      indexSizer += iItemSize;
+      length++;
+    }
+
+    /*
+     * Number of items to append at each end, i.e. 'preload' each side before entering view.
+     */
+    const bufferItems = Math.max(Math.floor(length / 4), 2);
+
+    /*
+     * This is how far we deviate into the bufferItems to detect a redraw.
+     */
+    const bufferSize = Math.max(Math.floor((length / 8) * defaultItemSize), 1);
+
+    const totalLength = length + bufferItems * 2 + 3;
+
+    setState({
+      virtualizerLength: totalLength,
+      virtualizerBufferSize: bufferSize,
+      virtualizerBufferItems: bufferItems,
+    });
+  };
+
+  // the handler for resize observer
+  const handleResize = debounce(resizeCallback);
+
+  // Keep the reference of ResizeObserver in the state, as it should live through renders
+  const [resizeObserver] = React.useState(canUseDOM() ? new ResizeObserver(handleResize) : undefined);
+
+  const scrollRef = React.useCallback(
+    (el: HTMLElement | null) => {
+      if (container.current !== el) {
+        if (container.current) {
+          resizeObserver?.unobserve(container.current);
+        }
+
+        container.current = el;
+        if (container.current) {
+          resizeObserver?.observe(container.current);
+        }
+      }
+    },
+    [resizeObserver],
+  );
+
+  if (container.current) {
+    const containerSize =
+      direction === 'vertical'
+        ? container.current?.getBoundingClientRect().height * 1.5
+        : container.current?.getBoundingClientRect().width * 1.5;
+
+    let couldBeSmaller = false;
+    let recheckTotal = 0;
+    for (let i = currentIndex; i < currentIndex + virtualizerLength; i++) {
+      const newItemSize = getItemSize(i);
+      sizeTracker.current[i] = newItemSize;
+      recheckTotal += newItemSize;
+
+      const newLength = i - currentIndex;
+
+      const bufferItems = Math.max(Math.floor(newLength / 4), 2);
+      const totalNewLength = newLength + bufferItems * 2 + 4;
+      const compareLengths = totalNewLength < virtualizerLength;
+
+      if (recheckTotal > containerSize && compareLengths && !couldBeSmaller) {
```

Comment:
```suggestion
      if (recheckTotal > containerSize && compareLengths) {
```

nit: it's rendundant I think?